selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/CheSeleniumMultiUserModule.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core;
+
+import com.google.inject.AbstractModule;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.client.CheTestMachineServiceClient;
+import org.eclipse.che.selenium.core.client.CheTestUserServiceClient;
+import org.eclipse.che.selenium.core.client.KeycloakTestAuthServiceClient;
+import org.eclipse.che.selenium.core.client.TestAuthServiceClient;
+import org.eclipse.che.selenium.core.client.TestMachineServiceClient;
+import org.eclipse.che.selenium.core.client.TestUserServiceClient;
+import org.eclipse.che.selenium.core.requestfactory.TestCheDefaultUserHttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.requestfactory.TestUserHttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.user.AdminTestUser;
+import org.eclipse.che.selenium.core.user.CheAdminTestUser;
+import org.eclipse.che.selenium.core.user.CheDefaultTestUser;
+import org.eclipse.che.selenium.core.user.CheTestUserNamespaceResolver;
+import org.eclipse.che.selenium.core.user.TestUser;
+import org.eclipse.che.selenium.core.user.TestUserNamespaceResolver;
+import org.eclipse.che.selenium.core.workspace.CheTestWorkspaceUrlResolver;
+import org.eclipse.che.selenium.core.workspace.TestWorkspaceUrlResolver;
+
+/** @author Anton Korneta */
+public class CheSeleniumMultiUserModule extends AbstractModule {
+
+  @Override
+  protected void configure() {
+    bind(TestAuthServiceClient.class).to(KeycloakTestAuthServiceClient.class);
+
+    bind(TestUser.class).to(CheDefaultTestUser.class);
+    bind(AdminTestUser.class).to(CheAdminTestUser.class);
+
+    bind(TestUserServiceClient.class).to(CheTestUserServiceClient.class);
+    bind(HttpJsonRequestFactory.class).to(TestUserHttpJsonRequestFactory.class);
+    bind(TestUserHttpJsonRequestFactory.class).to(TestCheDefaultUserHttpJsonRequestFactory.class);
+    bind(TestMachineServiceClient.class).to(CheTestMachineServiceClient.class);
```

Comment:
There is mess with class names.
`CheTest`, `Test`, `TestChe`

Reply:
Let me explain to Anton the naming scheme:
1. Usually we add **Test** prefix to the classes which are specific to selenium tests like the following:
> AuthServiceClient -> **Test**AuthServiceClient
> HttpJsonRequestFactory -> **Test**UserHttpJsonRequestFactory
2. In case of different implementation for Che, Codenvy Onprem, Keycloak, Codenvy SaaS, we add name of product at start of class name:
> TestAuthServiceClient -> **Che**TestAuthServiceClient
or -> **Onpem**TestAuthServiceClient
or -> **Keycloak**TestAuthServiceClient

According to this scheme, correct name of class 
> Test**Che**DefaultUserHttpJsonRequestFactory > **Che**TestDefaultUserHttpJsonRequestFactory.

Reply:
ok I'll fix that